libraries/AP_Scripting/lua_bindings.cpp

```diff
@@ -1270,4 +1271,28 @@ int lua_DroneCAN_get_FlexDebug(lua_State *L)
 }
 #endif // HAL_ENABLE_DRONECAN_DRIVERS
 
+#if AP_GPS_ENABLED
+int lua_gps_inject_data(lua_State *L)
+{
+    binding_argcheck(L, 3);       // self + 2 args
+    luaL_checkudata(L, 1, "gps"); // require GPS object as first argument
+
+    const char *data = luaL_checkstring(L, 2);
+    const uint16_t len = get_uint16_t(L, 3);
+
+    if (data == nullptr)
+    {
+        return luaL_error(L, "No data to inject");
+    }
+
+    if (len > 0)
+    {
+        AP::gps().inject_data((const uint8_t *)data, len);
+    }
```

Comment:
```suggestion
    binding_argcheck(L, 2);       // self + 1 arg
    luaL_checkudata(L, 1, "gps"); // require GPS object as first argument

    size_t len;
    const char *data = luaL_checklstring(L, 2, &len);

    if (len > 0 && len <= UINT16_MAX)
    {
        AP::gps().inject_data((const uint8_t *)data, (uint16_t)len);
    }
```

Please do it this way. Otherwise the user can pass a large length and potentially crash the autopilot. Lua is safe to have zeros in strings so I can't think of a reason to make the length an extra argument. `luaL_checklstring` cannot return a null pointer either, it raises a Lua error instead.

Reply:
You beat me to it. I just rebase/squashed and probably should've held off on pushing. Literally wrote what you have above in a pending commit, nearly verbatim! Thanks for the sanity check.